crates/svm-cli/src/wasm_value.rs

```diff
@@ -0,0 +1,74 @@
+use std::str::FromStr;
+
+use svm_app::types::WasmValue;
+
+#[derive(Debug, PartialEq)]
+pub enum ParseError {
+    InvalidFormat,
+    UnsupportedType(String),
+}
+
+/// Potentially converts `&str` to `WasmValue`.
+/// Expected format is `{num}{type}` (`type` in (i32|i64)).
+pub fn parse_str(value: &str) -> Result<WasmValue, ParseError> {
```

Comment:
a more idiomatic way would be to implement:

```rust
use std::convert::TryFrom;

impl TryFrom<&str> for WasmValue {
  type Error = ParseError;

  fn try_from(value: &str) -> Result<Self, Self::Error> {
     # ...
  }
}
```

Reply:
This was my go-to, but I got err `E0117`:
```
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types

   | impl TryFrom<&str> for WasmValue {
   | ^^^^^-------------^^^^^---------
   | |    |                 |
   | |    |                 `svm_app::types::wasm_value::WasmValue` is not defined in the current crate
   | |    `str` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead
```
I tried a few workarounds including using type alias and type redefine, without any success.

Reply:
my bad, `WasmValue` is defined in another crate.